marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -155,10 +155,11 @@ private JsonNode determineIngestionEndpointParams(Map<String, String> options) {
 
         if (doesNotHaveApiPath) {
             String apiPath = "/marklogic-data-hub-spark-connector/bulkIngester.api";
+            String scriptPath = "/marklogic-data-hub-spark-connector/bulkIngester.sjs";
 
-            if(hubClient.getModulesClient().newJSONDocumentManager().exists(apiPath) == null) {
+            if (endpointExists(apiPath)) {
```

Comment:
Need to check both the api and the sjs separately - it's conceivable that one exists but the other does not.

Reply:
Checking both the files.